packages/admin/admin/src/common/toolbar/DataGridToolbar.tsx

```diff
@@ -1,72 +1,73 @@
+import { type ComponentsOverrides, Paper, type PaperProps } from "@mui/material";
 import { css, type Theme, useThemeProps } from "@mui/material/styles";
-import { useGridApiContext } from "@mui/x-data-grid";
+import { type GridDensity, useGridApiContext } from "@mui/x-data-grid";
 
 import { createComponentSlot } from "../../helpers/createComponentSlot";
-import { type ThemedComponentBaseProps } from "../../helpers/ThemedComponentBaseProps";
-import { Toolbar, type ToolbarProps } from "./Toolbar";
 
-export type DataGridToolbarClassKey = "root" | "standard" | "comfortable" | "compact";
+export type DataGridToolbarClassKey = "root" | GridDensity;
 
-export type DataGridToolbarProps = {
-    /** @deprecated The `density` prop is deprecated. The density is now used from the DataGrid. */
-    density?: "standard" | "comfortable" | "compact";
-} & Omit<ToolbarProps, "slotProps" | "scopeIndicator" | "hideTopBar"> &
-    ThemedComponentBaseProps<{ root: typeof Toolbar }>;
+export type DataGridToolbarProps = PaperProps;
 
 type OwnerState = {
-    density: "standard" | "comfortable" | "compact";
+    density: GridDensity;
 };
 
-const Root = createComponentSlot(Toolbar)<DataGridToolbarClassKey, OwnerState>({
+export const DataGridToolbar = (inProps: DataGridToolbarProps) => {
+    const { elevation = 1, ...restProps } = useThemeProps({ props: inProps, name: "CometAdminDataGridToolbar" });
+    const apiRef = useGridApiContext();
+    const gridDensity = apiRef.current.state.density;
+
+    const ownerState: OwnerState = {
+        density: gridDensity,
+    };
+
+    return <Root ownerState={ownerState} elevation={elevation} {...restProps} />;
+};
+
+const Root = createComponentSlot(Paper)<DataGridToolbarClassKey, OwnerState>({
     componentName: "DataGridToolbar",
     slotName: "root",
     classesResolver(ownerState) {
         return [ownerState.density];
     },
 })(
     ({ ownerState, theme }) => css`
-        [class*="MuiDataGrid-toolbarQuickFilter"] {
-            width: 120px;
+        position: relative;
+        z-index: 1;
+        display: flex;
+        align-items: center;
+        gap: ${theme.spacing(2)};
+        padding: ${theme.spacing(2)};
 
+        ${ownerState.density === "comfortable" &&
+        css`
             ${theme.breakpoints.up("sm")} {
-                width: 150px;
+                padding-top: ${theme.spacing(4)};
+                padding-bottom: ${theme.spacing(4)};
             }
+        `}
 
-            ${theme.breakpoints.up("md")} {
-                width: "auto";
-            }
+        .CometAdminToolbarItem-root,
+        .CometAdminToolbarActions-root {
+            // Hide components that were required by the old usage of DataGridToolbar but would now break the styling.
+            // This is for usages of components that use ToolbarItem internally, such as ToolbarTitleItem or for existing projects where the usage was not updated.
+            display: contents;
```

Comment:
This seems like a workaround to me. Can't we update ToolbarTitleItem?

Reply:
> This seems like a workaround to me.

true.

> Can't we update ToolbarTitleItem?

I'm not sure if using `ToolbarTitleItem` or `ToolbarBackButton` (same issue) is actually a valid use-case inside a DataGrid's toolbar.
 
Will check with UX if we want to continue to support this. 
If we do, maybe we could check inside each of these components, if it is used inside a DataGrid and conditionally render the component with or without the `ToolbarItem` wrapper. Although that's still a workaround imo 🤔